### PR TITLE
Remove prefix from indexes of DynamoDB

### PR DESCRIPTION
**Pull request by yito88**
https://scalar-labs.atlassian.net/browse/DLT-8207

For DyanmoDB recovery, the prefix for index tables is removed.

Scalar DB requires the same name about index tables after restoring the table. However, the name of the restored tables should be different from that of the source table to restore data. This requirement can be satisfied by adding the prefix. Also, the name of an index table couldn't be modified after restoring the table on DyanmoDB.
That's why I don't want to add a prefix to the index.

### CODE REVIEW

File: src/main/java/com/scalar/db/api/Operation.java [hunk 1]
```diff
@@ -94,13 +94,28 @@ public Operation(Key partitionKey, Key clusteringKey) {
...
+    return forFullTableName(true);
+  }
+
+  /**
+   * Returns the full table name without the prefix with the namespace for this operation
+   *
+   * @return an {@code Optional} with the returned the full table name
+   */
+  @Nonnull
+  public Optional<String> forFullTableNameWithoutPrefix() {
```
Comment (feeblefakie) [inline src/main/java/com/scalar/db/api/Operation.java:106]:
```suggestion
  public Optional<String> forUnprefixedFullTableName() {
```
This sounds a bit better to me.

File: src/main/java/com/scalar/db/api/Operation.java [hunk 2]
```diff
@@ -94,13 +94,28 @@ public Operation(Key partitionKey, Key clusteringKey) {
    */
   @Nonnull
   public Optional<String> forFullTableName() {
+    return forFullTableName(true);
+  }
+
+  /**
+   * Returns the full table name without the prefix with the namespace for this operation
```
Comment (feeblefakie) [inline src/main/java/com/scalar/db/api/Operation.java:101]:
```suggestion
   * Returns the full table name without the unprefixed namespace for this operation
```

**Review by feeblefakie — COMMENTED** [review]
Looking good! Thank you!
Left minor naming suggestion.

**Comment by brfrn169** [comment]
@yito88 So an index name is a table-local name? I mean, there is no conflict with the same index names between the original table and the restored table?

File: src/main/java/com/scalar/db/api/Operation.java (hunk 1 above)
Comment (yito88) [reply src/main/java/com/scalar/db/api/Operation.java]: Thanks! Fixed.

**Comment by yito88** [comment]
@brfrn169 Good question! An index table is bound to the table. There is no conflict with that of the source table.

**Review by brfrn169 — APPROVED** [review]
@yito88 Thank you! LGTM!

**Review by feeblefakie — APPROVED** [review]
LGTM! Thank you!

**Pull request merged**